packages/next/client/image.tsx

```diff
@@ -276,6 +292,40 @@ export default function Image({
   if (!configEnableBlurryPlaceholder) {
     placeholder = 'empty'
   }
+  const isStatic = typeof src === 'object'
+  let staticSrc = ''
+  if (isStatic) {
+    let staticImport = src as StaticImport
+    let staticImageData: StaticImageData
+    if ((staticImport as StaticRequire).default) {
+      staticImageData = (staticImport as StaticRequire).default
+    } else {
+      staticImageData = staticImport as StaticImageData
+    }
+    if (!staticImageData.src) {
+      throw new Error(
+        `An object should only be passed to the image component src parameter if it comes from a static image import. It must include src. Received ${JSON.stringify(
+          staticImageData
+        )}`
+      )
+    }
+    if (staticImageData.dataURI) {
+      blurDataURL = staticImageData.dataURI
+    }
+    staticSrc = staticImageData.src
+    if (!layout || layout !== 'fill') {
+      height = staticImageData.height
+      width = staticImageData.width
+      if (!height || !width) {
+        throw new Error(
+          `An object should only be passed to the image component src parameter if it comes from a static image import. It must include height and width. Received ${JSON.stringify(
+            staticImageData
+          )}`
+        )
+      }
+    }
+  }
+  src = (isStatic ? staticSrc : src) as string
```

Comment:
```suggestion
  src = staticSrc || src
```

Will this work?

Reply:
Without this casting, a lot of references to `src` further down the file have problems because they only want a string. Since as of this line, we know `src` is a string no matter what, I think the cast is probably ok.

Reply:
```suggestion
  src = typeof src === ‘string’ ? src : staticSrc
```